airbyte-server/src/main/java/io/airbyte/server/handlers/SourceDefinitionsHandler.java

```diff
@@ -80,6 +130,42 @@ public SourceDefinitionReadList listSourceDefinitions() throws ConfigNotFoundExc
     return new SourceDefinitionReadList().sourceDefinitions(reads);
   }
 
+  public SourceDefinitionReadList listLatestSourceDefinitions() throws ConfigNotFoundException, IOException, JsonValidationException {
+    final JsonNode deserialize;
+    try {
+      deserialize = Yamls.deserialize(getLatestSources());
+      checkYamlIsPresentWithNoDuplicates(deserialize);
+    } catch (RuntimeException e) {
+      throw new KnownException(500, "Error retrieving latest source definitions", e);
+    }
+
+    final var sourceDefs = toStandardSourceDefinitions(deserialize.elements());
+    final var reads = sourceDefs.stream()
+        .map(SourceDefinitionsHandler::buildSourceDefinitionRead).collect(Collectors.toList());
+    return new SourceDefinitionReadList().sourceDefinitions(reads);
+  }
+
+  private String getLatestSources() {
```

Comment:
Same comment here regarding duplication and potential consolidation into 'one' class.